### PR TITLE
Add more unit tests for trainer, fix __init__.py and rename train.py to avoid import conflict

### PR DESCRIPTION
**Pull request by yangaws**
*Issue #, if available:*

*Description of changes:*
1, Add more unit tests for trainer
2, Fix __all__
3, Rename train.py to avoid import names conflict

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/tf_container/__init__.py [hunk 1]
```diff
@@ -11,8 +11,8 @@
 #  express or implied. See the License for the specific language governing 
 #  permissions and limitations under the License.
 
-from tf_container.train import train
+from tf_container.train_entry_point import train
```
Comment (mvsusp) [inline src/tf_container/__init__.py]: Where is train_entry_point coming from?

File: src/tf_container/__init__.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
...
 from tf_container.serve import load_dependencies, transformer
 import tf_container.serve as serve
 
-__all__ = [train, transformer, serve, load_dependencies]
+__all__ = ['train', 'transformer', 'serve', 'load_dependencies']
```
Comment (mvsusp) [inline src/tf_container/__init__.py]: Why is this change necessary?

File: test/unit/test_trainer.py [hunk 3]
```diff
@@ -67,20 +74,59 @@ def trainer(trainer_module):
                                  customer_params=HYPERPARAMETERS.copy())
 
 
+def test_test_trainer_params_passing(trainer_module):
```
Comment (winstonaws) [inline test/unit/test_trainer.py]: test_test_?

File: src/tf_container/__init__.py (hunk 1 above)
Comment (winstonaws) [reply src/tf_container/__init__.py]: This change renames train.py to train_entry_point.py, since otherwise there is no way to access the module tf_container.train. __init__.py imports the train function, which overrides the train module.

This affects our ability to test other functions from train_entry_point.py.

File: src/tf_container/__init__.py (hunk 2 above)
Comment (yangaws) [reply src/tf_container/__init__.py]: __all__ should include list of names instead of values.

From python doc(https://docs.python.org/2/tutorial/modules.html):

'The import statement uses the following convention: if a package’s __init__.py code defines a list named __all__, it is taken to be the list of module names that should be imported when from package import * is encountered. It is up to the package author to keep this list up-to-date when a new version of the package is released. Package authors may also decide not to support it, if they don’t see a use for importing * from their package. For example, the file sound/effects/__init__.py could contain the following code:

__all__ = ["echo", "surround", "reverse"]'

File: test/unit/test_trainer.py (hunk 3 above)
Comment (yangaws) [reply test/unit/test_trainer.py]: Fixed. Thanks

**Comment by yangaws** [comment]
@winstonaws 

New updates:
1, Added unit test for _resolve_value_for_training_input_fn_parameter functionality
2, I think we already have mock tests for _configure_s3_file_system(). I just updated a little bit to check the order.
